core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -283,25 +287,35 @@ public void start(Listener<RespT> observer, Metadata.Headers headers) {
       try {
         transport = obtainActiveTransport();
       } catch (RuntimeException ex) {
-        stream = new NoopClientStream();
-        listener.closed(Status.INTERNAL.withDescription("Failed starting transport").withCause(ex),
-            new Metadata.Trailers());
+        closeCallPrematurely(listener,
```

Comment:
maybe set `stream = new NoopClientStream()` at the top of this method before things can go wrong? Then you don't need `closeCallPrematurely`.